testsuite/forge/src/backend/k8s/prometheus.rs

```diff
@@ -1,22 +1,71 @@
 // Copyright (c) Aptos
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::Result;
+use crate::{create_k8s_client, Result};
 
 use anyhow::bail;
 use aptos_logger::info;
+use k8s_openapi::api::core::v1::Secret;
+use kube::api::Api;
 use prometheus_http_query::{response::PromqlResult, Client as PrometheusClient};
 use reqwest::{header, Client as HttpClient};
 use std::collections::BTreeMap;
 
-pub fn get_prometheus_client() -> Result<PrometheusClient> {
-    let prom_url =
-        std::env::var("PROMETHEUS_URL").unwrap_or_else(|_| "http://127.0.0.1:9090".to_string());
-    info!("Attempting to create prometheus client with: {} ", prom_url);
+pub async fn get_prometheus_client() -> Result<PrometheusClient> {
+    // read from the environment
+    let kube_client = create_k8s_client().await;
+    let secrets_api: Api<Secret> = Api::namespaced(kube_client.clone(), "default");
+
+    let prom_url_env = std::env::var("PROMETHEUS_URL");
+    let prom_token_env = std::env::var("PROMETHEUS_TOKEN");
+
+    let (prom_url, prom_token) = match (prom_url_env.clone(), prom_token_env) {
+        // if both variables are provided, use them, otherwise try inferring from environment
+        (Ok(url), Ok(token)) => {
+            info!("Creating prometheus client from environment variables");
+            (url, Some(token))
+        }
+        _ => {
```

Comment:
This behavior is worth unittesting

Reply:
fosho -- just added some unit tests to catch the case of (1) k8s secret (2) no k8s secret, but with partial env vars (3) no k8s secret, but with all env vars. Didn't care to test things like invalid prometheus URL, as those should panic and failfast for signal